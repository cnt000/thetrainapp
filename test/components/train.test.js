import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Train from '../../src/components/train'

const mock = {
  serviceIdentifier: 'W35422',
  serviceOperator: 'SW',
  transportMode: 'TRAIN',
  scheduledInfo: {
    scheduledTime: '2017-12-03T20:39:00+00:00',
    scheduledPlatform: '17'
  },
  callingType: 'PickUp',
  destinationList: [
    {
      crs: 'SNS'
    }
  ],
  realTimeUpdatesInfo: {
    realTimeServiceInfo: {
      realTime: '2017-12-03T20:39:00+00:00',
      realTimePlatform: '17',
      realTimeFlag: 'Estimate'
    }
  },
  callingPatternUrl:
    'https://realtime.thetrainline.com/callingPattern/W35422/2017-12-03'
}

describe('Train', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Train data={mock} onClick={() => console.log('clicked')} />,
      div
    )
  })

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Train data={mock} onClick={() => console.log('clicked')} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
