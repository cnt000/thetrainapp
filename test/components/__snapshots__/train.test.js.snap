// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Train has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa hPftyW"
>
  <div
    className="sc-bwzfXH fqjkKR"
  >
    21:39
  </div>
  <div
    className="sc-bxivhb vwWGx"
  >
    <div
      className="sc-bdVaJa hPftyW"
    >
      <span>
        SNS
      </span>
      <span>
        <abbr
          title="Platform"
        >
          Plat. 
        </abbr>
        17
      </span>
    </div>
    <div
      className="sc-bdVaJa hPftyW"
    >
      <span>
        W35422
        -
        SW
      </span>
      <span>
        On Time
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat bMTwEo"
  >
    <test-file-stub
      height={16}
      width={16}
    />
  </div>
</div>
`;
